code/analyticsedge/src/phone/java/com/adobe/marketing/mobile/AnalyticsExtension.java

```diff
@@ -435,4 +518,18 @@ ExecutorService getExecutor() {
             return executorService;
         }
     }
+
+    /**
+     * Getter for the {@link #executorService}. Access to which is mutex protected.
+     *
+     * @return A String containing a build application identifier
+     */
+    private String getApplicationIdentifier() {
+        final String applicationName =  systemInfoService.getApplicationName();
```

Comment:
It's quite a code execution, Do you think we can make it a instance property, if it needs to be use multiple times. 
Probably we can initialize it lazily.